Feature/styling

ixboy opened the pull request:
Hello, this is my BookStore project, this is the last milestone with the following requirements :

    You will find all the details of the Bookstore design in Zeplin.
    Login to Zeplin (hint: you will find the credentials in the sneak peek of the project).
    Open the project Bookstore CMS.
    Use the information detailed in Zeplin to style the website.
    Your final design should match Zeplin's design.

I have implemented all the requirements, please do review my work and tell me if I can proceed to the next one.

Thank You

ixboy committed:
add @mui/material

ixboy committed:
style App component

ixboy committed:
style form component

ixboy committed:
style bookList component

ixboy committed:
style category component

ixboy committed:
style components

ixboy committed:
remove unused code

ixboy committed:
fix linter error

ixboy committed:
add screenshot image to Readme file

omar-labana commented on src/components/App.css:
```diff
@@ -1 +1,13 @@
-@import url('https://fonts.googleapis.com/css2?family=Poppins:wght@100&family=Roboto:wght@100;400;700&display=swap');
+@import url('https://fonts.googleapis.com/css2?family=Montserrat:wght@300;700&family=Roboto+Slab&display=swap');
+
+body {
+  margin: 0;
+  padding: 0;
+  font-family: 'Roboto Slab', serif;
+  background-color: rgb(243, 243, 243);
+}
+
+.montserrat {
+  font-family: 'Montserrat', sans-serif;
+  font-size: 0.813rem;
+}
```
- [x] The app layout should match the Zeplin design, kindly make sure that all of your elements have a similar padding/margins.
Zeplin Design
![image](https://user-images.githubusercontent.com/33767951/142711703-c8772ee7-dc59-46af-b67a-8c0360c3f92e.png)
Your Styling
![image](https://user-images.githubusercontent.com/33767951/142711711-b3c6420b-94e0-416d-9540-27b0ba29b3cf.png)

omar-labana commented on src/components/App.js:
```diff
@@ -3,6 +3,7 @@ import {
   Route,
   Routes,
 } from 'react-router-dom';
+import './App.css';
```
- [x] The header (navbar) is missing the `avatar` icon, kindly add it.
- [x] Please make sure that the styling of the header matches the design.

Zeplin Design
![image](https://user-images.githubusercontent.com/33767951/142711840-4e86d45e-8d54-4847-bfbf-7a3a603f4c4e.png)

Your Styling
![image](https://user-images.githubusercontent.com/33767951/142711813-53f39c1c-4843-4cc9-95cb-3914324989c4.png)

ixboy committed:
remove links underline

omar-labana commented on src/components/BookList.js:
```diff
@@ -11,43 +13,52 @@ const BookList = () => {
+                <li className="text-primary fw-light mx-2">Edit</li>
               </ul>
             </div>
           </div>
-          <div className="px-3">
-            <div>Animated Circle</div>
-            <div>XX%</div>
-            <h4>Completed</h4>
+          <div className="d-flex flex-row px-3">
+            <div className="m-2"><CircularProgress color="primary" variant="determinate" value={progress} /></div>
+            <div className="flex-column">
+              <h2>
+                {progress}
+                %
+              </h2>
+              <h5 className="completed">Completed</h5>
+            </div>
```
- [x] Each two book should have some space between them, I would suggest copying the font styles from the design for the `book` component so it matches it (`font-size` and `font-weight`).
- [x] The vertical separator `|` between `progress` and `chapter info` is missing please add it so it matches the design.

Zeplin Design
![image](https://user-images.githubusercontent.com/33767951/142711938-fb734295-22f0-4eac-9f5a-3ad26a054da2.png)

Your Styling
![image](https://user-images.githubusercontent.com/33767951/142711861-7e7c1cf7-d9de-4d30-9e9e-f7eebed39b80.png)

omar-labana commented on src/components/BookForm.js:
```diff
@@ -16,21 +22,42 @@ const AddBook = () => {
-          <option value="test_2">Programing</option>
-          <option value="test_3">Movies</option>
-          <option value="test_4">Novels</option>
+        <input type="text" id="titleId" value={booktitle} onChange={addTitle} style={myStyle2} placeholder="Book title" />
+        <input type="text" id="authorId" value={bookauthor} onChange={addAuthor} style={myStyle2} placeholder="Book Author" />
+        <select type="select" className="p-0 w-25" id="category" style={myStyle2}>
+          <option value="Sci-Fi">Sci-fi</option>
+          <option value="Learning">Learning</option>
+          <option value="Programming">Programming</option>
+          <option value="Drama">Drama</option>
+          <option value="Thriller">Thriller</option>
+          <option value="Action">Action</option>
```
- [x] There should be a `<hr>` element between `BookList` and `BookForm`, please add it.
- [x] Kindly make sure to update the styling of your form (button and input fields) to match the given design.
- [x] The `author` name is found in your form but not in each `book`, kindly update it (I would also suggest removing the author input field while keeping the author's name in each `book` to match the design). 

Zeplin design
![image](https://user-images.githubusercontent.com/33767951/142712135-4582a78d-ef36-456b-ae53-71108220fff8.png)

Your Styling
![image](https://user-images.githubusercontent.com/33767951/142712148-1ce4638f-bf92-4722-89f0-b8d75a2a539e.png)

omar-labana reviewed (changes requested):
Hi @ixboy, 

Excellent work so far! 👌
You have implemented almost all the required features. ✔️ 
There are some cases that you need to adjust but you are almost done! 😀

## Required Changes ♻️

- [x] Please update the pull request info since this is not the `Math_Magicians` project.

- [x] _Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

ixboy committed:
add user avatar

ixboy committed:
remove bootstrap classes

ixboy committed:
refactor Navbar code and add avatar

ixboy committed:
delete some bookList styles

ixboy committed:
refactor bookform, change classes and elements

ixboy committed:
refactor all styles

ixboy committed:
remove bootstrap

ixboy committed:
add line div

ixboy committed:
change different classes

ixboy committed:
remove unused code

ixboy committed:
remove circularprogress and change variables and add other classes

ixboy committed:
edit all styles and fix bugs

ixboy committed:
fix linter errors

ixboy commented:
Hello and Thank You so much for the previous review, I have implemented all the required changes suggested. However I would like to highlight that I have removed the `Author input` from the form and also from the `BookList`  Because the form in the design does not have the author's name input, and I wanted to much the design.

Thank You again for the previous review...

D4nielJ commented on src/components/BookList.js:
```diff
@@ -11,43 +11,57 @@ const BookList = () => {
+              Chapter
+              {` ${chapter}`}
+            </p>
+            <button className="Update-progress" type="button">Update Progress</button>
```
[OPTIONAL] I think you should reset the styles of the buttons to get rid of the default borders from the browsers. It would make your project looks much better and similar to the template.

ixboy replied:
Thank You very Much for the review, I will make that change now.  :)

D4nielJ reviewed (approved):
## Hello @ixboy, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉

![](https://i.imgur.com/NLzCKYy.gif)

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear or you can always find me on slack!

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

ixboy committed:
remove buttons borders
